sources/api/schnauzer/tests/multicall_integration.rs

```diff
@@ -0,0 +1,230 @@
+        Command::new(binary_path)
+            .args(args)
+            .output()
+            .expect(&format!(
+                "Failed to execute {:?} with args {:?}",
+                binary_path, args
+            ))
+    }
+
+    /// Check if the output contains v1-specific usage message
+    fn is_v1_output(&self, output: &Output) -> bool {
+        let stderr = String::from_utf8_lossy(&output.stderr);
+        stderr.contains("Usage:") && stderr.contains("SETTING_KEY")
+    }
+
+    /// Check if the output is from v2
+    fn is_v2_output(&self, output: &Output) -> bool {
+        !self.is_v1_output(output)
+    }
+}
+
+/// Test that the multicall binary can be invoked directly as schnauzer
+#[test]
+fn test_direct_schnauzer_invocation() {
+    // Given A multicall binary built by cargo
+    let test = MulticallTest::new();
+
+    // When Invoking the binary directly with --help
+    let output = test.run_command(Path::new(test.binary_path), &["--help"]);
+
+    // Then Should exit with usage error code for v1 behavior
+    assert_eq!(output.status.code(), Some(USAGE_ERROR_EXIT_CODE));
+
+    // Should contain usage information
+    let stderr = String::from_utf8_lossy(&output.stderr);
+    assert!(stderr.contains("Usage:"));
+}
+
+/// Test that the multicall binary can be invoked directly as schnauzer-v2
+#[test]
+fn test_direct_schnauzer_v2_invocation() {
+    // Given A multicall binary copied with schnauzer-v2 name
+    let test = MulticallTest::new();
+    let v2_binary_path = test.create_copy("schnauzer-v2");
+
+    // When Invoking the renamed binary with --help
+    let output = test.run_command(&v2_binary_path, &["--help"]);
+
+    // Then v2 should handle --help differently than v1
+    assert!(output.status.success() || output.status.code() == Some(2));
+    assert!(test.is_v2_output(&output));
+}
+
+/// Test symlink behavior - the core multicall functionality
+#[test]
+fn test_symlink_dispatch() {
+    // Given A multicall binary and symlinks with different names
+    let test = MulticallTest::new();
+    let schnauzer_link = test.create_symlink("schnauzer");
+    let schnauzer_v2_link = test.create_symlink("schnauzer-v2");
+
+    // When Invoking each symlink with --help
+    let output_v1 = test.run_command(&schnauzer_link, &["--help"]);
+    let output_v2 = test.run_command(&schnauzer_v2_link, &["--help"]);
+
+    // Then Each should route to the correct implementation
+    assert!(
+        test.is_v1_output(&output_v1),
+        "schnauzer should route to v1"
+    );
+    assert!(
+        test.is_v2_output(&output_v2),
+        "schnauzer-v2 should route to v2"
+    );
+
+    // The outputs should be different, indicating different code paths
+    let stderr_v1 = String::from_utf8_lossy(&output_v1.stderr);
+    let stderr_v2 = String::from_utf8_lossy(&output_v2.stderr);
+    assert_ne!(
+        stderr_v1, stderr_v2,
+        "v1 and v2 should produce different outputs"
+    );
```

Comment:
The additional checks are only useful because `is_v2_output` just negates the result of `is_v1_output`. 

If `is_v2_output` actually looked for v2-specific output, like v1 does, then this test could more concisely be written as:
```
    assert!(
        test.is_v1_output(&output_v1),
        "schnauzer should route to v1"
    );

    assert_ne!(
        test.is_v2_output(&output_v1),
        "v1 output should not be classified as v2"
    );

    assert!(
        test.is_v2_output(&output_v2),
        "schnauzer-v2 should route to v2"
    );

    assert_ne!(
        test.is_v1_output(&output_v2),
        "v2 output should not be classified as v1"
    );
```